### PR TITLE
feat: Add multidirectional latency with target selectors for precise directional control

### LINKED ISSUE #14: Adjust latencies between regions
### Feature to Enhance

The current netem chaos testing use one latency for both directions.

### Requested Enhancement

We need the latency (actually the round trip time) to emulate the average round trip time between regions which are:
- us <-> us: 20ms
- us <-> ap: 200ms
- us <-> eu: 100ms
- eu <-> eu: 20ms
- eu <-> ap: 300ms
- ap <-> ap: 20ms

### Business Need

To represent the actual latency between consensus nodes across the globe

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by brunodam**
## Description

This pull request introduces a major overhaul to the network chaos testing framework, focusing on more realistic global latency emulation and improved diagnostics capabilities. The changes standardize the configuration of network latency between regions, replace previous ad-hoc scenarios with a comprehensive global latency matrix, and add new tasks and resources for cluster diagnostics and cleanup. Resource naming is now fixed to prevent duplication and latency accumulation across runs.

**Network Chaos Emulation Improvements**
* Replaced previous region scenarios with a global latency matrix, specifying round-trip times (RTT) between US, EU, and AP regions for more accurate network emulation. All NetworkChaos resources now use fixed names instead of UUIDs, ensuring subsequent runs replace existing resources and avoid compounding latency. (`README.md`, `chaos/network/netem-800ms.yml`, `chaos/network/netem-ap-melbourne.yml`, `chaos/network/netem-eu-london.yml`, `chaos/network/netem-us-ohio.yml`) [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L80-R183) [[2]](diffhunk://#diff-13b84ceff53676ad051e583fa8fb18bbc904622036d1c4d2c68758b8fe258ae4L4-R4) [[3]](diffhunk://#diff-c3fb60fe3ca50373bed309dd3d6556193303b7123d630cf74be424bbc425e86eL4-R4) [[4]](diffhunk://#diff-c3fb60fe3ca50373bed309dd3d6556193303b7123d630cf74be424bbc425e86eR15-R88) [[5]](diffhunk://#diff-6adef8f08cbc10dd8362770e37d53ff00463d08084e6d90b78d224f96dda6167L4-R4) [[6]](diffhunk://#diff-6adef8f08cbc10dd8362770e37d53ff00463d08084e6d90b78d224f96dda6167R15-R86) [[7]](diffhunk://#diff-d265a5bfde8170ad0f008e078513604c357b634766366bcd0e1d198ff0c407b1L4-R4) [[8]](diffhunk://#diff-d265a5bfde8170ad0f008e078513604c357b634766366bcd0e1d198ff0c407b1R15-R84)

**Task Automation and Cleanup**
* Updated `Taskfile.chaos.network.yml` to automate the application of global latency scenarios, show experiment status for each fixed resource, and provide new tasks for deploying, cleaning up, and interacting with cluster diagnostics pods. Added a cleanup task for NetworkChaos experiments. (`chaos/Taskfile.chaos.network.yml`)

**Cluster Diagnostics Pod Enhancements**
* Added new tasks and documentation for deploying, accessing, and cleaning up cluster diagnostics pods, which now reside in the `solo` namespace. The diagnostics pod is equipped with a suite of network testing tools and can be deployed with region-specific labels for targeted testing. (`README.md`, `chaos/Taskfile.chaos.network.yml`, `dev/k8s/cluster-diagnostics.yaml`) [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R247-R249) [[2]](diffhunk://#diff-98dfb660cc249ef575bdfe524f7615f1c9a1b758381428978e7f24b97d77bdaaL30-R96) [[3]](diffhunk://#diff-75356c6e1254db213e03ee47e7b484210aa31306eb0a662ff97a9d00e6dd30bdL3-R6) [[4]](diffhunk://#diff-75356c6e1254db213e03ee47e7b484210aa31306eb0a662ff97a9d00e6dd30bdL36-L52) [[5]](diffhunk://#diff-75356c6e1254db213e03ee47e7b484210aa31306eb0a662ff97a9d00e6dd30bdL63-L137) [[6]](diffhunk://#diff-75356c6e1254db213e03ee47e7b484210aa31306eb0a662ff97a9d00e6dd30bdL155-R63)

**Kubernetes Resource Standardization**
* Consolidated cluster diagnostics services into a single headless service (`cluster-diagnostics-svc`) in the `solo` namespace, removing individual node services and the separate namespace for diagnostics. All supporting resources (ConfigMap, ServiceAccount, etc.) are now scoped to `solo`. (`dev/k8s/cluster-diagnostics.yaml`) [[1]](diffhunk://#diff-75356c6e1254db213e03ee47e7b484210aa31306eb0a662ff97a9d00e6dd30bdL3-R6) [[2]](diffhunk://#diff-75356c6e1254db213e03ee47e7b484210aa31306eb0a662ff97a9d00e6dd30bdL63-L137) [[3]](diffhunk://#diff-75356c6e1254db213e03ee47e7b484210aa31306eb0a662ff97a9d00e6dd30bdL155-R63)

**Documentation Updates**
* Expanded and clarified documentation for network chaos testing and diagnostics, including usage examples, technical implementation details, and the rationale for resource naming and region configuration. (`README.md`) [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L80-R183) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R247-R249)

Let me know if you want to dive deeper into any specific change or need help understanding how to use the new tasks!

### Related Issues

* Closes #14 

### Pull request (PR) checklist

* \[ ] This PR added tests (unit, integration, and/or end-to-end)
* \[x] This PR updated documentation
* \[x] This PR added no TODOs or commented out code
* \[ ] This PR has no breaking changes
* \[ ] Any technical debt has been documented as a separate issue and linked to this PR
* \[ ] Any `go.mod` and/or `go.sum` changes have been explained to and approved by a repository manager
* \[ ] All related issues have been linked to this PR
* \[ ] All changes in this PR are included in the description
* \[ ] When this PR merges the commits will be squashed and the title will be used as the commit message, the 'commit
  message guidelines' below have been followed

### Testing

* \[ ] This PR added unit tests
* \[ ] This PR added integration/end-to-end tests
* \[x] These changes required manual testing that is documented below
* \[ ] Anything not tested is documented

The following manual testing was done:

* TBD

The following was not tested:

* TBD

<details>
<summary>
Commit message guidelines
</summary>
We use 'Conventional Commits' to ensure that our commit messages are easy to read, follow a consistent format, and for automated release note generation. Please follow the guidelines below when writing your commit messages:

1. BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a
   breaking API change (correlating with MAJOR in Semantic Versioning). A BREAKING CHANGE can be part of commits of any
   type. NOTE: currently breaking changes will only bump the MAJOR version.
2. The title is prefixed with one of the following:

| Prefix    | Description                                         | Semantic Version Update | Captured in Release Notes |
|-----------|-----------------------------------------------------|-------------------------|---------------------------|
| feat:     | a new feature                                       | MINOR                   | Yes                       |
| fix:      | a bug fix                                           | PATCH                   | Yes                       |
| perf:     | performance                                         | PATCH                   | Yes                       |
| refactor: | code change that isn't feature or fix               | none                    | No                        |
| test:     | adding missing tests                                | none                    | No                        |
| docs:     | changes to documentation                            | none                    | Yes                       |
| build:    | changes to build process                            | none                    | No                        |
| ci:       | changes to CI configuration                         | none                    | No                        |
| style:    | formatting, missing semi-colons, etc                | none                    | No                        |
| chore:    | updating grunt tasks etc; no production code change | none                    | No                        |

</details>

### CODE REVIEW

File: dev/k8s/cluster-diagnostics.yaml [hunk 1]
```diff
@@ -179,133 +87,13 @@ roleRef:
...
-            - name: entrypoint
-              configMap:
-                name: cluster-diagnostics-cm
----
-apiVersion: apps/v1
-kind: Deployment
-metadata:
-  name: cluster-diagnostics-node4
-  namespace: cluster-diagnostics
+  name: cluster-diagnostics
```
Comment (brunodam) [inline dev/k8s/cluster-diagnostics.yaml:95]: Leaving just one deployment for cluster-diagnostics for the moment given we've being just using ping to check RTT for now.

File: dev/k8s/cluster-diagnostics.yaml [hunk 2]
```diff
@@ -33,23 +28,6 @@ data:
...
-      aarch64|arm64)
-        ARCH="arm64"
-        ;;
-      *)
-        echo "FATAL: Unsupported architecture: ${ARCH}"
-        exit 1
-        ;;
-    esac
-    
-    ${SUDO} curl -Lo /usr/local/bin/kubectl "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/${ARCH}/kubectl"
```
Comment (brunodam) [inline dev/k8s/cluster-diagnostics.yaml:50]: detection of architecture needed for installation of curl is failing. Given we are not using curl for now, just simplifying this bit as well.

**Review by leninmehedy — APPROVED** [review]

**Review by leninmehedy — APPROVED** [review]
